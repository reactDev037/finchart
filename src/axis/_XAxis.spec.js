import React from 'react';
import { scaleLinear } from 'd3-scale';

import XAxis from './_XAxis';

describe('<_XAxis />', () => {
  it('renders ticks as child elements correctly', () => {
    const width = 900;
    const height = 360;
    const margin = 20;
    const xScale = scaleLinear().range([0, width]).domain([0, 100]);
    const axis = render(<XAxis {...{ xScale, height, margin }} />);
    expect(axis).toMatchSnapshot();
  });

  it('renders the group element correctly', () => {
    const width = 900;
    const height = 360;
    const margin = 20;
    const xScale = scaleLinear().range([0, width]).domain([0, 100]);
    const axis = render(<XAxis {...{ xScale, height, margin }} />);
    expect(axis.find('g').prop('transform')).toBe(`translate(0,${height})`);
  });

  it('sets a hashed class name', () => {
    const width = 900;
    const height = 360;
    const xScale = scaleLinear().range([0, width]).domain([0, 100]);
    const axis = render(<XAxis {...{ xScale, height }} />);
    expect(axis.find('g').prop('class')).toBeDefined();
  });
});
