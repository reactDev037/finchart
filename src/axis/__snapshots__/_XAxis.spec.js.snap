// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<_XAxis /> renders ticks as child elements correctly 1`] = `
<g
  class="sc-bdVaJa cNoDXm"
  transform="translate(0,360)"
>
  <text
    class="label"
    transform="translate(450, 40)"
  />
</g>
`;
